services/weblate/weblate-entity-count.service.js

```diff
@@ -11,64 +10,46 @@ const schema = Joi.object({
 }).required()
 
 const queryParamSchema = Joi.object({
-  server: optionalUrl.required(),
+  server: optionalUrl,
 }).required()
 
-class WeblateEntityCountBase extends BaseJsonService {
+module.exports = class WeblateEntityCount extends BaseJsonService {
   static category = 'other'
-
-  static buildRoute(entityName) {
-    return {
-      base: 'weblate',
-      pattern: entityName,
-      queryParamSchema,
-    }
+  static route = {
+    base: 'weblate/count',
```

Comment:
I think we should consider using a slightly more descriptive noun here, both because that'd align better with the class and file name, and also due to the terminology overlap with the user statistics.

Is my understanding correct that these data points reflective of the entire server/instance, whereas the user stats are specific to that particular user on the specified server/instance? I guess phrased a different way, what's the difference this `languages` badge vs the user stats `languages` badge?

My sense is that the entity counts would essentially be the aggregate of _all_ user counts, and if that's the case it may make sense to use consistent terminology (i.e.`count` vs. `statistics`). We don't have to structure our routes based on the upstream API route, so if one of those (or even something different) will make more sense as the noun for our consumers we should consider that.

At a minimum, I'd think we'd need something like this given the current routes:

 ```suggestion
    base: 'weblate/entity-count',
```

Reply:
Sorry about that!

> Is my understanding correct that these data points reflective of the entire server/instance, whereas the user stats are specific to that particular user on the specified server/instance

Correct! Or just to be 100% we're on the same page:

This route is aggregate stats of the entire server instance.

* projects - Is the total number of projects across the entire Weblate instance created by any user.
* languages - Is the total number of languages enabled across all projects. (ie if a server has 2 projects, 1 has `en`, `fr`, `es`, and the other has `de`, `en-GB`, then it'd be 5 in total)
* users - Is the total number of users on the entire Weblate instance. 

Still happy with `entity-count`, or would you think something else would be better?

Reply:
> Still happy with entity-count, or would you think something else would be better?

Not so much. It's still unclear to me why something like `languages` is an `entity` when in aggregate but a `user` (statistic) when more granular. Ultimately, the noun really is the entity and the fact that the quantity of said noun is displayed is fairly implicit, much in the way that we don't use the noun `download-count` for our download badges.

It really feels like it should be more like:

* `/weblate/:noun`, e.g. `/weblate/languages`
* `/weblate/:noun/:user`, e.g. `/weblate/languages/nijel`

Where the former is inherently the total and the latter inherently specific to/filtered by the provided user.

For the other user stats, we also need to be consistent with the verbiage since they are actually the noun portion (e.g. `commented` -> `comments`, `uploaded` -> `uploads`, etc.). Any requisite mapping to the upstream API values should happen internally within the service class

Reply:
Ahh, I see yeah. 🤔 
Now that I think about it, I'm an idiot.

In the Wikiapiary PR, I was encouraged to use `:variant` as the first part of the path. I guess I can just do that here as well.
I was thinking I shouldn't use a parameter for the noun.

I'll update this now.